### DISCUSSION TITLE
v1.4.0 Is there anyway of sending shape hints to MutliFormatWriter?

### DISCUSSION THREAD

**Discussion by StenBone** [zxing-cpp/zxing-cpp#519] [Q&A] [answered]
Hello,
I want to directly use `setShapeHint` from DMWriter.h. As far as I there is no way to access `setShapeHint` from the PUBLIC_HEADERS API. 
Why are the Data Matrix headers not included as part of the public API?
Can shape hints be set through `MultiFormatWriter`?

**Comment by axxel** [comment, marked as answer]
> Why are the Data Matrix headers not included as part of the public API?

This was done to reduce the public API to allow for future refactoring.

> Can shape hints be set through `MultiFormatWriter`?

Unfortunately not. This can be considered an oversight. If you need to get access, you could simply copy the missing header to where your client code can see it. The symbols themselves have not been touched / removed.

Please note that the writer API will change in the future. See #332.
